src/plots/plots.js

```diff
@@ -1632,7 +1632,13 @@ plots.autoMargin = function(gd, id, o) {
             delete pushMarginIds[id];
         }
         else {
-            var pad = o.pad === undefined ? 12 : o.pad;
+            var pad = o.pad;
+            if(pad === undefined) {
+                var margin = fullLayout.margin;
+                // if no explicit pad is given, use 12px unless there's a
+                // specified margin that's smaller than that
+                pad = Math.min(12, margin.l, margin.r, margin.t, margin.b);
+            }
```

Comment:
@etpinard seem reasonable? I started looking into this due to test failures at https://github.com/plotly/plotly.js/pull/2681/commits/d02c76b3d1082996fa1c2543d7c8f08e08a256c0 - where some colorbars that were previously not able to expand the margin because they were internally pixel-sized suddenly became able to expand it when I switched it to plot-fraction-sized as described in https://github.com/plotly/plotly.js/pull/2681#discussion_r192409090... but then this 12px padding overrode user-supplied margins that were smaller - and that didn't seem right to me. Some of the baseline images with small explicit margins changed as a result.

The 12px here is an odd hard-coded feature - I suppose an alternative would be to make this an attribute in its own right (`layout.margin.autopad`?) and put the `min(12, margin.l...)` logic in `supplyDefaults`.

Reply:
> seem reasonable?

Very reasonable. :1st_place_medal: 

> I suppose an alternative would be to make this an attribute in its own right (layout.margin.autopad?) and put the min(12, margin.l...) logic in supplyDefaults.

Maybe down the road, but no need in this PR.